### DISCUSSION TITLE
Unified alerting: Support provisioning

### DISCUSSION THREAD

**Discussion by tom0010** [grafana/grafana#39510] [Alerting]
Slack thread:
https://grafana.slack.com/archives/C0Y4TLW74/p1632134449337500 

Summary: 
We'd like to have json rules support for the new unified alerting. So we can predefine a set of rules prior to building a grafana instance, very similar in the way you can do this with dashboards and data sources.

For me personally, I run multiple grafana instances in multiple sites for geo redundancy, so having the same rules applied along with deduplication for the new unified alerting will be a huge win. 
By deduplication, I mean to not have the same alert fired off twice or three times because I have grafana deployed in a another site, matching the same rule set. Vanilla alertmanager handles this, if grafana alertmanager doesn't, can this be added too?

**Comment by ismarslomic** [comment]
Just confirming that this feature is very important for us as well and is currently a show stopper for migrating existing alert rules from Prometheus Alertmanager to Grafana Alerts. 

Without being able to export alerting rules (with corresponding settings done in UI) as JSON, to be maintained in version control per environment we can't go further in our migration.

Is this feature on the roadmap? Any input that could help us planning the next steps would be appreciated!

**Reply by tom0010** [reply]
https://github.com/grafana/grafana/pull/37920

**Reply by ismarslomic** [reply]
Tnx for sharing, however we are not talking about setting up Grafana in HA in this thread. Could be something Im missing, but not sure the config you refer to in the commit will work in our case

**Reply by ismarslomic** [reply]
Any update on this issue?

**Reply by peterholmberg** [reply]
Hello! Yes provisioning alert rules (I think that's what you're asking for?) is on the roadmap for next year.

**Reply by Nkmol** [reply]
@peterholmberg Thanks for this notion. Do you maybe already have an issue on this GitHub that might track this process and on which I can subscribe to? This also the thing that is blocking us from using the new alerts set up.

**Reply by snevesbarros** [reply]
@peterholmberg  same as the question above I would like to understand where provisioning alerts is in the roadmap. Thanks.

**Reply by pssa** [reply]
> @peterholmberg same as the question above I would like to understand where provisioning alerts is in the roadmap. Thanks.

I second this.

**Reply by hardik-fareye** [reply]
Hello @peterholmberg 
Can you please share the expected time of release for the "provisioning alert rules" feature? It would be really helpful.

**Reply by danfinn** [reply]
Also curious what the timeline is on this as this is holding us back from moving to unified alerting.

**Reply by rmawhinnie** [reply]
has this been prioritized, this is blocking our adoption of unified alerting

**Reply by cjs226** [reply]
Interested as well as this is a blocker for us to opt-in to unified.

**Reply by dougalII** [reply]
This is an issue for us as well

**Comment by gotjosh** [comment]
Hey All, I have renamed this discussion to `Support provisioning` to better reflect what's being requested here. This is currently under active development and should land just around Grafana 9.

As for the high availability without duplication, this is already built-in as part of grafana https://grafana.com/docs/grafana/next/alerting/unified-alerting/high-availability/

**Comment by ievgen-golubiev** [comment]
@gotjosh Do you have any plans on this issue?

**Comment by JohnnyQQQQ** [comment]
We are currently working on file and terraform provisioning of alert rules (https://github.com/grafana/grafana/issues/36153#issuecomment-1087597922). The goal is to land it in 9.1. As an alternative one can already build their own provisioning around our new provisioning API that is docuemented here https://grafana.com/docs/grafana/latest/developers/http_api/alerting_provisioning/

**Reply by gotjosh** [reply]
Thanks @JohnnyQQQQ, locking the thread as I don't think there's anything further to add here.
